### PR TITLE
GC unused StringEnum, and avoid including config in header files.

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge or @geirst PR

### CODE REVIEW

**Review by geirst — APPROVED** [review]
👍

**Review by toregge — COMMENTED** [review]
:+1:

File: searchcore/src/vespa/searchcore/proton/docsummary/isummarymanager.h [hunk 1]
```diff
@@ -20,6 +18,8 @@ namespace proton {
 class ISummaryManager
 {
 public:
+    using SummaryConfig = vespa::config::search::internal::InternalSummaryType;
```
Comment (toregge) [inline searchcore/src/vespa/searchcore/proton/docsummary/isummarymanager.h:21]: Consider adding const, to differentiate between config builder (writeable) and config (read only).

Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/docsummary/isummarymanager.h:21]: Fixed

**Pull request merged**